go/test/endtoend/keyspace/keyspace_test.go

```diff
@@ -0,0 +1,338 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sequence
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	"vitess.io/vitess/go/vt/proto/topodata"
+)
+
+var (
+	clusterForKSTest      *cluster.LocalProcessCluster
+	keyspaceShardedName   = "test_ks_sharded"
+	keyspaceUnshardedName = "test_ks_unsharded"
+	cell                  = "zone1"
+	cell2                 = "zone2"
+	hostname              = "localhost"
+	servedTypes           = map[topodata.TabletType]bool{topodata.TabletType_MASTER: true, topodata.TabletType_REPLICA: true, topodata.TabletType_RDONLY: true}
+	sqlSchema             = `create table vt_insert_test (
+								id bigint auto_increment,
+								msg varchar(64),
+								keyspace_id bigint(20) unsigned NOT NULL,
+								primary key (id)
+								) Engine=InnoDB`
+	shardKIdMap = map[string][]uint64{
+		"-80": {527875958493693904, 626750931627689502,
+			345387386794260318, 332484755310826578,
+			1842642426274125671, 1326307661227634652,
+			1761124146422844620, 1661669973250483744,
+			3361397649937244239, 2444880764308344533},
+		"80-": {9767889778372766922, 9742070682920810358,
+			10296850775085416642, 9537430901666854108,
+			10440455099304929791, 11454183276974683945,
+			11185910247776122031, 10460396697869122981,
+			13379616110062597001, 12826553979133932576},
+	}
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	exitCode := func() int {
+		clusterForKSTest = &cluster.LocalProcessCluster{Cell: cell, Hostname: hostname}
+		defer clusterForKSTest.Teardown()
+
+		// Start topo server
+		if err := clusterForKSTest.StartTopo(); err != nil {
+			return 1
+		}
+
+		if err := clusterForKSTest.VtctlProcess.AddCellInfo(cell2); err != nil {
+			return 1
+		}
+
+		// Start sharded keyspace
+		keyspaceSharded := &cluster.Keyspace{
+			Name:      keyspaceShardedName,
+			SchemaSQL: sqlSchema,
+		}
+		if err := clusterForKSTest.StartKeyspace(*keyspaceSharded, []string{"-80", "80-"}, 1, false); err != nil {
+			return 1
+		}
+		if err := clusterForKSTest.VtctlclientProcess.ExecuteCommand("SetKeyspaceShardingInfo", "-force", keyspaceShardedName, "keyspace_id", "uint64"); err != nil {
```

Comment:
I believe this is v2. The v3 way is to use vschema.

Reply:
I could not find the vschema way to specify `ShardingColumnName` , it only takes `vindex` and `table`. Other way is to specify `sharding_column_name` at the time of `createkeyspace`, are you referring that to v3 way ?

Reply:
I mean that `SetKeyspaceShardingInfo` is itself v2. Try removing the calls to this command. I bet the test will fail at line 305. The way to fix it may be to define a vschema that shards on keyspace_id column using a numeric vindex.
But also, it may not be necessary to test this in keyspace_test at all, because with vschema, it is not the keyspace that is computing keyspace_id, so the keyspace_id functionality should be covered by the vtgatev3 test.

Reply:
Got it, in recent commit I implemented what you are suggesting and testcase is passing with the new change.